libindy/src/commands/metrics.rs

```diff
@@ -12,18 +29,585 @@ const OPENED_WALLET_IDS_COUNT: &str = "opened_wallet_ids_count";
+    DidCommandGetDidMetadata,
+    DidCommandAbbreviateVerkey,
+    DidCommandGetNymAck,
+    DidCommandGetAttribAck,
+    DidCommandQualifyDid,
+    // WalletCommand
+    WalletCommandRegisterWalletType,
+    WalletCommandCreate,
+    WalletCommandCreateContinue,
+    WalletCommandOpen,
+    WalletCommandOpenContinue,
+    WalletCommandClose,
+    WalletCommandDelete,
+    WalletCommandDeleteContinue,
+    WalletCommandExport,
+    WalletCommandExportContinue,
+    WalletCommandImport,
+    WalletCommandImportContinue,
+    WalletCommandGenerateKey,
+    WalletCommandDeriveKey,
+    // PairwiseCommand
+    PairwiseCommandPairwiseExists,
+    PairwiseCommandCreatePairwise,
+    PairwiseCommandListPairwise,
+    PairwiseCommandGetPairwise,
+    PairwiseCommandSetPairwiseMetadata,
+    // NonSecretsCommand
+    NonSecretsCommandAddRecord,
+    NonSecretsCommandUpdateRecordValue,
+    NonSecretsCommandUpdateRecordTags,
+    NonSecretsCommandAddRecordTags,
+    NonSecretsCommandDeleteRecordTags,
+    NonSecretsCommandDeleteRecord,
+    NonSecretsCommandGetRecord,
+    NonSecretsCommandOpenSearch,
+    NonSecretsCommandFetchSearchNextRecords,
+    NonSecretsCommandCloseSearch,
+    // PaymentsCommand
+    PaymentsCommandRegisterMethod,
+    PaymentsCommandCreateAddress,
+    PaymentsCommandCreateAddressAck,
+    PaymentsCommandListAddresses,
+    PaymentsCommandAddRequestFees,
+    PaymentsCommandAddRequestFeesAck,
+    PaymentsCommandParseResponseWithFees,
+    PaymentsCommandParseResponseWithFeesAck,
+    PaymentsCommandBuildGetPaymentSourcesRequest,
+    PaymentsCommandBuildGetPaymentSourcesRequestAck,
+    PaymentsCommandParseGetPaymentSourcesResponse,
+    PaymentsCommandParseGetPaymentSourcesResponseAck,
+    PaymentsCommandBuildPaymentReq,
+    PaymentsCommandBuildPaymentReqAck,
+    PaymentsCommandParsePaymentResponse,
+    PaymentsCommandParsePaymentResponseAck,
+    PaymentsCommandAppendTxnAuthorAgreementAcceptanceToExtra,
+    PaymentsCommandBuildMintReq,
+    PaymentsCommandBuildMintReqAck,
+    PaymentsCommandBuildSetTxnFeesReq,
+    PaymentsCommandBuildSetTxnFeesReqAck,
+    PaymentsCommandBuildGetTxnFeesReq,
+    PaymentsCommandBuildGetTxnFeesReqAck,
+    PaymentsCommandParseGetTxnFeesResponse,
+    PaymentsCommandParseGetTxnFeesResponseAck,
+    PaymentsCommandBuildVerifyPaymentReq,
+    PaymentsCommandBuildVerifyPaymentReqAck,
+    PaymentsCommandParseVerifyPaymentResponse,
+    PaymentsCommandParseVerifyPaymentResponseAck,
+    PaymentsCommandGetRequestInfo,
+    PaymentsCommandSignWithAddressReq,
+    PaymentsCommandSignWithAddressAck,
+    PaymentsCommandVerifyWithAddressReq,
+    PaymentsCommandVerifyWithAddressAck,
+    // CacheCommand
+    CacheCommandGetSchema,
+    CacheCommandGetSchemaContinue,
+    CacheCommandGetCredDef,
+    CacheCommandGetCredDefContinue,
+    CacheCommandPurgeSchemaCache,
+    CacheCommandPurgeCredDefCache,
+    // MetricsCommand
+    MetricsCommandCollectMetrics,
+    // Exit
+    Exit,
+}
```

Comment:
I think it would be better to move CommandIndex-related boilerplate to separate file to improve readability of core metrics logic. However this is just a matter of taste...